### PR TITLE
Adding Content to Debugging Sponge Within the IDE

### PR DESCRIPTION
**Pull request by Grauldon**
This PR is a continuation of #784 and adds content to the [Debugging Sponge Within the IDE page](https://docs.spongepowered.org/stable/en/contributing/implementation/debugging/ide.html). It also adds an HTML target for referencing the [Setup the Workspace](https://docs.spongepowered.org/stable/en/contributing/implementation/debugging/index.html#setup-the-workspace) section of the index page and cleans up some relative paths that are more than two deep.

Additional PRs will add more content in the future.

### CODE REVIEW

**Comment by Grauldon** [comment]
Requested review from ST-DDT to compare content with [comment](https://github.com/SpongePowered/SpongeDocs/pull/661#issuecomment-364664272) in #661.

> How to debug (at least some few lines , a link to tutorials for common IDEs or a link to a debug section for plugins)

- [x] post mixin class content output
- [x] Tools for decompiling the post mixing code
- [x] How to debug
- [x] Suggest the creation of a test plugin
- [x] Common issues - Broken mixin
- [ ] Common issues - NPEs
- [ ] Common issues - PhaseErrors
- [ ] Help with finding the error in your mixin based on the stacktrace

**Review by ST-DDT — APPROVED** [review]
I haven't tested the IDEA specific part, but this is exactly what I wanted. Thanks!

File: source/contributing/implementation/debugging/ide.rst [hunk 1]
```diff
@@ -2,26 +2,91 @@
...
-IntelliJ
---------
+    See the :ref:`runConfig` section and the following :ref:`usingDebugger` section on the Plugin Debugging page for 
+    more information on debugging with an IDE.
+
+    This article is currently based on IntelliJ. If you are an Eclipse user and feel like you can expand this article 
+    to include Eclipse, you can do so on `our GitHub repository <https://github.com/spongepowered/spongedocs>`_. 
+
+Using your IDE to debug Sponge is relatively straight-forward. For the most part, you import the ``build.gradle`` file 
+into your IDE after you have :ref:`setupWorkspace`. Ensure you have JDK 8 installed and specified in your IDE, and you
```
Comment (Inscrutable) [inline source/contributing/implementation/debugging/ide.rst]: the second "you" is redundant (i.e. "and import the project").

File: source/contributing/implementation/debugging/ide.rst [hunk 2]
```diff
@@ -2,26 +2,91 @@
...
+   with SpongeForge or ``SpongeVanilla_main`` when working with SpongeVanilla. 
+
+Using Your IDE
+==============
+
+Now with everything working, you can do all of the usual tasks when in your IDE, such as setting breakpoints, stepping 
+through code, examining variables, and evaluating expressions. You can also utilize regular aspects of the game, such 
+as commands and log output.
+
+Debugging With Mods And Plugins
```
Comment (Inscrutable) [inline source/contributing/implementation/debugging/ide.rst]: Correct Title Case would be "Debugging with Mods and Plugins". (https://titlecase.com/ is useful).

File: source/contributing/implementation/debugging/ide.rst [hunk 3]
```diff
@@ -2,26 +2,91 @@
...
+-------------------------------
+
+When debugging code with a mod or plugin, place a copy of the mod's or plugin's jar file in the ``run/mods`` directory.
```
Comment (Inscrutable) [inline source/contributing/implementation/debugging/ide.rst]: Rather than "the mod's or plugin's" can we have "it's" instead?

**Review by Inscrutable — COMMENTED** [review]
Naught but some minor quibbles. Full steam ahead, mister Grauldon!

File: source/contributing/implementation/debugging/ide.rst (hunk 1 above)
Comment (Grauldon) [reply source/contributing/implementation/debugging/ide.rst]: Corrected

File: source/contributing/implementation/debugging/ide.rst (hunk 2 above)
Comment (Grauldon) [reply source/contributing/implementation/debugging/ide.rst]: Corrected - Thanks for the link!

File: source/contributing/implementation/debugging/ide.rst (hunk 3 above)
Comment (Grauldon) [reply source/contributing/implementation/debugging/ide.rst]: Yes, a much better read for sure!
Corrected

**Review by ST-DDT — APPROVED** [review]
LGTM

**Pull request merged**